staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -746,3 +750,25 @@ func Warning(cmdErr io.Writer, newGeneratorName, oldGeneratorName string) {
 		oldGeneratorName,
 	)
 }
+
+var SubresourceTypes = sets.NewString("status", "scale")
+var StatusOnlySubresourceTypes = sets.NewString("status")
+
+func IsValidSubresource(cmd string, subresource string) error {
```

Comment:
@eddiezane can you explain more about this requested change. I am afraid I do not follow what the request is.
As you suggesting that each cmd have its own `IsValidSubresource` function?

Reply:
Sorry - should have been more clear.

Instead of having command specific validation in this helpers file we'd like to see this logic move to their commands or abstracted.

Reply:
now it is clear :) thanks.

Addressing all the changes now.